apps/spruce/src/hooks/useHasProjectOrRepoEditPermission/index.ts

```diff
@@ -0,0 +1,52 @@
+import { useQuery } from "@apollo/client";
+import {
+  IsRepoQuery,
+  IsRepoQueryVariables,
+  UserProjectSettingsPermissionsQuery,
+  UserProjectSettingsPermissionsQueryVariables,
+  UserRepoSettingsPermissionsQuery,
+  UserRepoSettingsPermissionsQueryVariables,
+} from "gql/generated/types";
+import {
+  IS_REPO,
+  USER_PROJECT_SETTINGS_PERMISSIONS,
+  USER_REPO_SETTINGS_PERMISSIONS,
+} from "gql/queries";
+
+export const useHasProjectOrRepoEditPermission = (id?: string) => {
+  const { data: isRepoData, loading } = useQuery<
+    IsRepoQuery,
+    IsRepoQueryVariables
+  >(IS_REPO, {
+    variables: { projectOrRepoId: id ?? "" },
+    skip: id === "",
```

Comment:
```suggestion
    skip: !id,
```
Since `id` can be `null` or `undefined`